lib/src/_shared/KeyboardDateInput.tsx

```diff
@@ -83,7 +80,7 @@ export const KeyboardDateInput: React.FC<DateInputProps & DateInputRefs> = ({
     if ((rawValue === null || utils.isValid(rawValue)) && !isFocused) {
       setInnerInputValue(getInputValue());
     }
-  }, [rawValue]); // eslint-disable-line
+  }, [rawValue, utils, inputFormat]); // eslint-disable-line
```

Comment:
```suggestion
  }, [rawValue, utils, inputFormat]);
```

Reply:
There is a side-effect vakye `isFocused` which should not be used as a dependency. Do you think we need to move it to refs?

Reply:
I don't follow. If we use `isFocused` in the logic, why not include it in the dependency? If it doesn't have a negative side effect, and if it allows us to removes `eslint-disable-line`. Then, it could help us catch issues after a refactoring.

A side note on the coding convention, we tends to avoid is as much as possible for variables. For instance:

- https://github.com/mui-org/material-ui/blob/fe687bf2a13c3cb158547edc5494b23029b59b0b/packages/material-ui-lab/src/useAutocomplete/useAutocomplete.js#L194
- https://github.com/mui-org/material-ui/blob/fe687bf2a13c3cb158547edc5494b23029b59b0b/packages/material-ui/src/InputBase/InputBase.js#L225

Reply:
We don't need to run effect if the input is changed state from focus to blur. We only need to handle it differently by the fact of focused state. 

So this focused state does not need to be state actually. It can be a ref
```jsx
const isFocusedRef = React.useRef(false) 

onFocus={() => isFocusedRef.current = true} 
onBlur={() => isFocusedRef.current = false} 
```

Reply:
+1 for using a ref if we don't need to rerender. 

I can see `emptyInputText` as a dependency that is ignored. Is this a dead prop?